mp/src/vgui2/vgui_controls/CheckButton.cpp

```diff
@@ -151,6 +151,17 @@ void CheckButton::SetSelected(bool state )
 	}
 }
 
+//-----------------------------------------------------------------------------
+// Purpose: Check the button without posting an action signal
+//-----------------------------------------------------------------------------
+void CheckButton::SilentSetSelected(bool state)
+{
+    if (m_bCheckButtonCheckable)
+    {
+        BaseClass::SetSelected(state);
+    }
```

Comment:
This is sharing the same code as set selected, why not just make a shared internal method that takes an extra parameter `bPostActionSignal`? That way both of these functions call that and there's no code duplication